processor/samplingprocessor/tailsamplingprocessor/idbatcher/id_batcher_test.go

```diff
@@ -143,18 +144,18 @@ func concurrencyTest(t *testing.T, numBatches, newBatchesInitialCapacity, batchC
 
 	idSeen := make(map[string]bool, len(ids))
 	for _, id := range got {
-		idSeen[string(id)] = true
+		idSeen[string(id.Bytes())] = true
```

Comment:
Doesn't look like your changes necessarily affect this but I saw this construct in a couple places and seems kinda odd. Is the id guaranteed to be valid utf-8 or no? Should this be using `idSeen[id.HexString()]` instead?

Reply:
>Is the id guaranteed to be valid utf-8 or no?

Not guaranteed. Are you asking because it is converted to a string? Strings in Go do not have to be utf8, they are just immutable byte sequences.